### PR TITLE
Improve the docker-compose setup

### PR DESCRIPTION
**Pull request by flavio**
**NOTE WELL:** not ready to be merged, I created this PR to collect feedback.

This PR addresses issue https://github.com/SUSE/Portus/issues/217. It makes incredibly easier to play with Portus.

This setup is intended for development purposes only.

By running the ./compose-setup.sh file the following operations are
going to happen:

  * the mariadb image will be pulled
  * the web image (containing Portus) will be built
  * the registry image (based on openSUSE:13.2) will be built
  * Portus' database will be properly configured
  * Further instructions will be printed

As you have guessed this new setup allows to run the both registry,
portus and mariadb inside of Docker containers.

[Live demo](https://asciinema.org/a/24174)

TODO:
  * rewrite the commit
  * write documentation

### CODE REVIEW

File: docker/registry/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM opensuse:13.2
```
Comment (lsamayoa) [inline docker/registry/Dockerfile]: Why not use the official registry image ?

Comment (flavio) [inline docker/registry/Dockerfile]: Fixed, it was a leftover

File: compose-setup.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+
+function setup_database() {
```
Comment (mssola) [inline compose-setup.sh]:
```bash
setup_database() {
```
It's recommended to declare functions without the `function` keyword for portability reasons.

File: compose-setup.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+echo "DOCKER_HOST=${HOST}" > docker/environment
+
+
+docker-compose kill
+docker-compose rm -fv
+docker-compose up -d
+
+setup_database
+
+echo "###################"
```
Comment (mssola) [inline compose-setup.sh]: Use a heredoc

**Comment by mssola** [comment]
I think that you can put the compose-common file into this new directory `docker`, I'd like to avoid putting more files in the root of the project :)

**Comment by mssola** [comment]
LGTM :+1:  !

Also, could you take care of updating the documentation on [this](https://github.com/SUSE/Portus/wiki/The-development-environment#docker) wiki section?

**Comment by flavio** [comment]
> Also, could you take care of updating the documentation on this wiki section?

Sure, I already planned about doing that.

**Comment by flavio** [comment]
Docs updated, I'll update the website now. Merging in the meantime.

**Pull request merged**